boards/riscv/hifive1/board.cmake

```diff
@@ -11,6 +11,8 @@ set(QEMU_FLAGS_${ARCH}
   -machine sifive_e
   )
 
-board_set_debugger_ifnset(qemu)
```

Comment:
Why is this line being removed?

Reply:
Run the `west debug`, and we get the following error.
It seems not to work currently.
```
FATAL ERROR: board hifive1 does not support runner qemu
```

This PR makes it able to debug with an actual hifive1 board.
I think it is good to make device debugging the default because it is a general behavior in zephyr.

Reply:
ack